lib/guest_installation_and_configuration_base.pm

```diff
@@ -393,6 +265,38 @@ sub prepare_common_environment {
     return $self;
 }
 
+=head2 prepare_ssh_key
+
+  prepare_ssh_key($self)
+
+Prepare ssh key [ssh_public_key] and [ssh_private_key] for passwordless ssh login
+from host to guest. [ssh_command] stores options and username to be used with ssh
+login.
+
+=cut
+
+sub prepare_ssh_key {
+    my $self = shift;
+
+    $self->reveal_myself;
+    if (!((script_run("[[ -f /root/.ssh/id_rsa.pub ]] && [[ -f /root/.ssh/id_rsa.pub.bak ]]") == 0) and (script_run("cmp /root/.ssh/id_rsa.pub /root/.ssh/id_rsa.pub.bak") == 0))) {
+        assert_script_run("rm -f -r /root/.ssh/id_rsa*");
+        assert_script_run("ssh-keygen -t rsa -f /root/.ssh/id_rsa -q -P \"\" <<<y");
+        assert_script_run("cp /root/.ssh/id_rsa.pub /root/.ssh/id_rsa.pub.bak");
+    }
+    $self->{ssh_public_key} = script_output("cat /root/.ssh/id_rsa.pub");
+    $self->{ssh_private_key} = script_output("cat /root/.ssh/id_rsa");
+    # Use 'qevirt' as ssh login name for SLE Micro 6.0 onwards, because ssh root
+    # login is not permitted at the moment. It is not trivial to enable root ssh
+    # login for image-based guest installation, and there will be no impact on
+    # verifying whether guest is successfully installed or not. So using non-root
+    # user 'qevirt' is fairly enough, reasonable and does no harm. Please refer
+    # to https://bugzilla.suse.com/show_bug.cgi?id=1217220.
+    my $_username = ($self->{guest_os_name} eq 'slem' and $self->{guest_version} >= 6.0) ? 'qevirt' : 'root';
```

Comment:
https://bugzilla.suse.com/show_bug.cgi?id=1217220#c41 is resolved. Configuring root ssh login via ignition or combustion can be done without much effort I think. Which part has high cost?
The main need of using root comes from more feature tests in future, which is highly likely I think.